rpc/src/rpc_service.rs

```diff
@@ -131,12 +135,9 @@ impl RpcRequestMiddleware {
             return false;
         }
 
-        let starting_character = '/';
-        if !path.starts_with(starting_character) {
-            return false;
-        }
-
-        let path = path.trim_start_matches(starting_character);
+        let Some(path) = Self::strip_prefix(path) else {
```

Comment:
nit: hehe, i like `let else` (https://blog.rust-lang.org/2022/11/03/Rust-1.65.0.html#let-else-statements) but you'll need extra work for backporting:

```
$ git grep -E '^.*stable_version=1|channel = ' origin/{master,v1.1{4,3}} -- ./rust-toolchain.toml ./ci/rust-version.sh 
origin/master:rust-toolchain.toml:channel = "1.66.1"
origin/v1.14:ci/rust-version.sh:  stable_version=1.60.0
origin/v1.13:ci/rust-version.sh:  stable_version=1.59.0
```